tests/meson.build

```diff
@@ -0,0 +1,113 @@
+# (This file requires ../examples/meson.build to be processed already.)
+
+
+if not meson.can_run_host_binaries()
+	subdir_done() # We are cross-compiling, but could not execute test binaries.
+endif
+
+
+if not find_program('/bin/sh', required : false).found()
+	subdir_done() # Required command interpreter missing.
+endif
+
+
+if not find_program('cmp', required : false).found()
+	subdir_done() # Required shell command not found.
+endif
+
+
+# The test scripts need $srcdir pointing to the directory of the 'test*.out' files.
+srcdir_param = 'srcdir='+meson.current_source_dir()
+
+
+tests = [
+	# <Name>     <test script>  <executable> 
+	[ 'test 1',  './test1.sh',  test1  ],
```

Comment:
No need for the `./` here, the final lookup looks like this:

```
Program ./test1.sh found: YES (/tmp/tmpi08csggu/src/tclap-1.2.1/./tests/./test1.sh)
```

meson already uses the full path.